### ISSUE TITLE
StringIndexOutOfBoundsException Index 1 out of bounds for length 1

### ISSUE THREAD

**Issue by chilampoon** [igvteam/igv#1357]
Hi, when I load BAM files from cellranger (short read scRNA-seq data aligned to GRCh38) in IGV, I usually encountered:

`Error encountered querying alignments: java.lang.StringIndexOutOfBoundsException: Index 1 out of bounds for length 1`

My laptop is MacBook pro with M1 Pro chip, I have tried IGV 2.16.1 with and without built-in java but got the same error with both.
If I open BAM files aligned from minimap2, I didn't get this error.
Any idea to resolve this problem? thanks

**Comment by jrobinso** [comment]
See https://github.com/igvteam/igv/issues/1343

**Comment by chilampoon** [comment]
hi @jrobinso so I removed the igv folder with  `prefs.properties` in it and the application, then reinstalled it, but I still got the same java error after I opened one file which is the same one that triggered this error before...

here is the log

```
INFO [Jul 13,2023 16:07] [Main] Startup  IGV Version 2.16.1 04/19/2023 02:19 AM
INFO [Jul 13,2023 16:07] [Main] Java 11.0.13 (build 11.0.13+8) 2021-10-19
INFO [Jul 13,2023 16:07] [Main] Java Vendor: Eclipse Adoptium https://adoptium.net/
INFO [Jul 13,2023 16:07] [Main] JVM: OpenJDK 64-Bit Server VM Temurin-11.0.13+8   
INFO [Jul 13,2023 16:07] [Main] OS: Mac OS X 13.4.1 x86_64
INFO [Jul 13,2023 16:07] [Main] IGV Directory: /Users/hhh123/igv
INFO [Jul 13,2023 16:07] [CommandListener] Listening on port 60151
INFO [Jul 13,2023 16:07] [GenomeManager] Loading genome: https://s3.amazonaws.com/igv.org.genomes/hg19/hg19.json
INFO [Jul 13,2023 16:07] [TrackLoader] Loading resource:  https://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/ncbiRefSeq.txt.gz
INFO [Jul 13,2023 16:07] [GenomeManager] Loading genome: https://s3.amazonaws.com/igv.org.genomes/hg38/hg38.json
INFO [Jul 13,2023 16:07] [TrackLoader] Loading resource:  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/ncbiRefSeq.txt.gz
INFO [Jul 13,2023 16:07] [TrackLoader] Loading resource:  /Users/hhh123/playground/test/kras/bam/sample6486994.possorted_genome_bam.bam
INFO [Jul 13,2023 16:09] [TrackLoader] Loading resource:  /Users/hhh123/playground/test/kras/bam/sample6497566.possorted_genome_bam.bam
INFO [Jul 13,2023 16:09] [TrackLoader] Loading resource:  /Users/hhh123/playground/test/kras/bam/sampleCC0930.possorted_genome_bam.bam
SEVERE [Jul 13,2023 16:09] [AlignmentTileLoader] Error loading alignment data
SEVERE [Jul 13,2023 16:09] [AlignmentTileLoader] java.lang.StringIndexOutOfBoundsException: String index out of range: 1
    at java.base/java.lang.StringLatin1.charAt(Unknown Source)
    at java.base/java.lang.String.charAt(Unknown Source)
    at org.igv/org.broad.igv.sam.mods.BaseModificationUtils.getBaseModificationSets(BaseModificationUtils.java:76)
    at org.igv/org.broad.igv.sam.SAMAlignment.getBaseModificationSets(SAMAlignment.java:338)
    at org.igv/org.broad.igv.sam.mods.BaseModificationCounts.incrementCounts(BaseModificationCounts.java:81)
    at org.igv/org.broad.igv.sam.BaseAlignmentCounts.incCounts(BaseAlignmentCounts.java:132)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader$AlignmentTile.addRecord(AlignmentTileLoader.java:494)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader.loadTile(AlignmentTileLoader.java:245)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.loadInterval(AlignmentDataManager.java:391)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.load(AlignmentDataManager.java:338)
    at org.igv/org.broad.igv.sam.CoverageTrack.load(CoverageTrack.java:186)
    at org.igv/org.broad.igv.ui.IGV.lambda$repaint$16(IGV.java:2267)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
    at java.base/java.lang.Thread.run(Unknown Source)

INFO [Jul 13,2023 16:09] [TrackLoader] Loading resource:  /Users/hhh123/playground/test/kras/bam/sampleCC0930.possorted_genome_bam.bam
SEVERE [Jul 13,2023 16:09] [AlignmentTileLoader] Error loading alignment data
SEVERE [Jul 13,2023 16:09] [AlignmentTileLoader] java.lang.StringIndexOutOfBoundsException: String index out of range: 1
    at java.base/java.lang.StringLatin1.charAt(Unknown Source)
    at java.base/java.lang.String.charAt(Unknown Source)
    at org.igv/org.broad.igv.sam.mods.BaseModificationUtils.getBaseModificationSets(BaseModificationUtils.java:76)
    at org.igv/org.broad.igv.sam.SAMAlignment.getBaseModificationSets(SAMAlignment.java:338)
    at org.igv/org.broad.igv.sam.mods.BaseModificationCounts.incrementCounts(BaseModificationCounts.java:81)
    at org.igv/org.broad.igv.sam.BaseAlignmentCounts.incCounts(BaseAlignmentCounts.java:132)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader$AlignmentTile.addRecord(AlignmentTileLoader.java:494)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader.loadTile(AlignmentTileLoader.java:245)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.loadInterval(AlignmentDataManager.java:391)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.load(AlignmentDataManager.java:338)
    at org.igv/org.broad.igv.sam.CoverageTrack.load(CoverageTrack.java:186)
    at org.igv/org.broad.igv.ui.IGV.lambda$repaint$16(IGV.java:2267)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
    at java.base/java.lang.Thread.run(Unknown Source)

INFO [Jul 13,2023 16:10] [ShutdownThread] Shutting down
INFO [Jul 13,2023 16:10] [Main] Startup  IGV Version 2.16.1 04/19/2023 02:19 AM
INFO [Jul 13,2023 16:10] [Main] Java 11.0.13 (build 11.0.13+8) 2021-10-19
INFO [Jul 13,2023 16:10] [Main] Java Vendor: Eclipse Adoptium https://adoptium.net/
INFO [Jul 13,2023 16:10] [Main] JVM: OpenJDK 64-Bit Server VM Temurin-11.0.13+8   
INFO [Jul 13,2023 16:10] [Main] OS: Mac OS X 13.4.1 x86_64
INFO [Jul 13,2023 16:10] [Main] IGV Directory: /Users/hhh123/igv
INFO [Jul 13,2023 16:10] [CommandListener] Listening on port 60151
INFO [Jul 13,2023 16:10] [GenomeManager] Loading genome: https://s3.amazonaws.com/igv.org.genomes/hg19/hg19.json
INFO [Jul 13,2023 16:10] [TrackLoader] Loading resource:  https://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/ncbiRefSeq.txt.gz
INFO [Jul 13,2023 16:10] [GenomeManager] Loading genome: /Users/hhh123/igv/genomes/hg38.json
INFO [Jul 13,2023 16:10] [TrackLoader] Loading resource:  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/ncbiRefSeq.txt.gz
INFO [Jul 13,2023 16:10] [TrackLoader] Loading resource:  /Users/hhh123/playground/test/kras/bam/sampleCC0930.possorted_genome_bam.bam
SEVERE [Jul 13,2023 16:10] [AlignmentTileLoader] Error loading alignment data
SEVERE [Jul 13,2023 16:10] [AlignmentTileLoader] java.lang.StringIndexOutOfBoundsException: String index out of range: 1
    at java.base/java.lang.StringLatin1.charAt(Unknown Source)
    at java.base/java.lang.String.charAt(Unknown Source)
    at org.igv/org.broad.igv.sam.mods.BaseModificationUtils.getBaseModificationSets(BaseModificationUtils.java:76)
    at org.igv/org.broad.igv.sam.SAMAlignment.getBaseModificationSets(SAMAlignment.java:338)
    at org.igv/org.broad.igv.sam.mods.BaseModificationCounts.incrementCounts(BaseModificationCounts.java:81)
    at org.igv/org.broad.igv.sam.BaseAlignmentCounts.incCounts(BaseAlignmentCounts.java:132)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader$AlignmentTile.addRecord(AlignmentTileLoader.java:494)
    at org.igv/org.broad.igv.sam.AlignmentTileLoader.loadTile(AlignmentTileLoader.java:245)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.loadInterval(AlignmentDataManager.java:391)
    at org.igv/org.broad.igv.sam.AlignmentDataManager.load(AlignmentDataManager.java:338)
    at org.igv/org.broad.igv.sam.CoverageTrack.load(CoverageTrack.java:186)
    at org.igv/org.broad.igv.ui.IGV.lambda$repaint$16(IGV.java:2267)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
    at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by jrobinso** [comment]
Did you read #1343?  This is a 10X problem, out-of-spec BAM files.  I have a workaround to allow reading these for the next release.   See the #1343 for more info.

**Comment by chilampoon** [comment]
oh okay, i thought it'd been the 'next release'... good to know, thanks!

**Comment by jrobinso** [comment]
If you need it now you can use the snapshot build. https://software.broadinstitute.org/software/igv/download_snapshot.   The next release will be 2.17.0, by the end of the month.

**Comment by jrobinso** [comment]
@chilampoon We have released a patch for this today, version 2.16.2, available on the downloads page

**Comment by chilampoon** [comment]
hi @jrobinso just tested 2.16.2, the problem is resolved, thanks!
